db/attachment.go

```diff
@@ -455,30 +435,22 @@ func (db *Database) ForEachStubAttachment(body Body, minRevpos int, callback Att
 	if atts == nil && body["_attachments"] != nil {
 		return base.HTTPErrorf(400, "Invalid _attachments")
 	}
-	for name, value := range atts {
-		meta, ok := value.(map[string]interface{})
-		if !ok {
-			return base.HTTPErrorf(400, "Invalid attachment")
-		}
-		if meta["data"] == nil {
-			if revpos, ok := base.ToInt64(meta["revpos"]); revpos < int64(minRevpos) || !ok {
+	for name, meta := range atts {
+		if meta.Data == nil {
+			if meta.Revpos < minRevpos {
```

Comment:
Previous base.ToInt64 would have returned ok=false if meta["revpos"] was nil.  The above comparison doesn't seem equivalent, as the uninitialized version of meta.Revpos will be zero.  

To more easily preserve the existing logic, maybe we want to change to *int/*bool/*string in the DocAttachment struct?